src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -45,9 +45,11 @@ public void onCompletion(RecordMetadata md, Exception e) {
 		if ( e != null ) {
 			if ( e instanceof RecordTooLargeException ) {
 				LOGGER.error("RecordTooLargeException @ " + position + " -- " + key);
-				LOGGER.error("Maxwell dropped a row because it was too large (" + e.getLocalizedMessage() + ")");
+				LOGGER.error(e.getLocalizedMessage());
 				LOGGER.error("Considering raising max.request.size broker-side.");
-				inflightMessages.completeMessage(position);
+
+				if ( isTXCommit )
+					inflightMessages.completeMessage(position);
```

Comment:
don't you need to set the new position in `context`? (similar to what is done below)